clang/test/CodeGen/aie/looppragma.c

```diff
@@ -0,0 +1,60 @@
+// --- looppragma.c ---------------------------------------------------------///
+// NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+// --------------------------------------------------------------------------///
+// RUN: %clang --target=aie2 -S -emit-llvm %s -o - | FileCheck %s
```

Comment:
Maybe also run that test at -O2 to verify the metadata is preserved? I guess for the last test it will not, at least without the next commit.

Reply:
I had intended this to be a unit test for the frontend

Reply:
Yes let's keep it that way, I see you added tests for -O2 in the next commit.